### PR TITLE
Add documentation for dev helm charts

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
###### Description

Add documentation for dev helm charts. 

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/docs/Dev.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+e.g.
+
+```bash
+helm upgrade collection sumologic-dev/sumologic \
+        --install \
+        --namespace sumologic \
+        --version 2.0.0-dev.0-81-g07e8e27f \
+        -f /sumologic/vagrant/values.yaml,/sumologic/vagrant/values.local.yaml
+```
```
Comment (pmalek-sumo) [inline deploy/docs/Dev.md:34]: Add empty line at the end

File: deploy/docs/Dev.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Development
+
+This document contains information helpful for developers.
+
+## Installation of non-official Helm charts
+
+Non-official Helm charts are available in [dev](https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/gh-pages/dev) directory on [gh-pages](https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/gh-pages) branch.
```
Comment (pmalek-sumo) [inline deploy/docs/Dev.md:7]: Fancy markdown tricks I learnt recently

```suggestion
Non-official Helm charts are available in [dev] directory on [gh-pages] branch.

[dev]: https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/gh-pages/dev
[gh-pages]: https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/gh-pages
```

Comment (sumo-drosiek) [reply deploy/docs/Dev.md]: @pmalek-sumo shouldn't those links be stored at the end of the file? 🤔

**Review by sumo-drosiek — APPROVED** [review]
:+1:

File: deploy/docs/Dev.md (hunk 2 above)
Comment (pmalek-sumo) [reply deploy/docs/Dev.md]: Matter of preference I guess 🤷🏻

**Review by perk-sumo — APPROVED** [review]
Thank you! 🚀

**Pull request merged**